pkg/metrics_store/metrics_writer.go

```diff
@@ -82,13 +87,22 @@ func (m MetricsWriter) WriteAll(w io.Writer) error {
 	return nil
 }
 
-// SanitizeHeaders removes duplicate headers from the given MetricsWriterList for the same family (generated through CRS).
-// These are expected to be consecutive since G** resolution generates groups of similar metrics with same headers before moving onto the next G** spec in the CRS configuration.
-func SanitizeHeaders(writers MetricsWriterList) MetricsWriterList {
+// SanitizeHeaders sanitizes the headers of the given MetricsWriterList.
+func SanitizeHeaders(contentType string, writers MetricsWriterList) MetricsWriterList {
 	var lastHeader string
 	for _, writer := range writers {
 		if len(writer.stores) > 0 {
 			for i, header := range writer.stores[0].headers {
+				// If the requested content type was proto-based, replace the type with "gauge", as "info" and "statesets" are not recognized by Prometheus' protobuf machinery.
```

Comment:
Use a good function name to replace comments.

Wrap these into a function.
```
				if strings.HasPrefix(contentType, expfmt.ProtoFmt) &&
					strings.HasPrefix(header, "# TYPE") &&
					(strings.HasSuffix(header, " "+string(metric.Info)+"\n") || strings.HasSuffix(header, " "+string(metric.StateSet)+"\n")) {
					typeStringWithoutTypePaddedIndex := strings.LastIndex(header, " ")
					typeStringWithoutType := header[:typeStringWithoutTypePaddedIndex]
					writer.stores[0].headers[i] = typeStringWithoutType + string(metric.Gauge)
				}
```

Reply:
Wrapping both of the behaviors defined here might impact the readability. Additionally, for eg., if those functions use a helper function, that would mean context switching between three different functions to figure out what's going on. In contrast, the function length doesn't exceed the one enforced by the linters, but we could consider that once that happens.

I think we'd want to include a rule for it in the `golangci` config if we really want to consistently check for patterns such as this.